File: app/src/main/java/com/shentu/wallpaper/app/page/EmptyCallback.java
package com.shentu.wallpaper.app.page;

import com.kingja.loadsir.callback.Callback;
import com.shentu.wallpaper.R;


public class EmptyCallback extends Callback {

    private int emptyResId;


    public EmptyCallback(int emptyResId) {
        this.emptyResId = emptyResId;
    }

    public EmptyCallback() {
        this.emptyResId = R.layout.layout_default_empty;
    }

    @Override
    protected int onCreateView() {
        return emptyResId;
    }

}

File: app/src/main/java/com/shentu/wallpaper/app/page/EmptyCallback.kt
package com.shentu.wallpaper.app.page

import com.kingja.loadsir.callback.Callback
import com.shentu.wallpaper.R

class EmptyCallback : Callback {
    private var emptyResId: Int

    constructor(emptyResId: Int) {
        this.emptyResId = emptyResId
    }

    constructor() {
        emptyResId = R.layout.layout_default_empty
    }

    override fun onCreateView(): Int {
        return emptyResId
    }
}